numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -1028,7 +1060,22 @@ PyArray_IntpFromIndexSequence(PyObject *seq, npy_intp *vals, npy_intp maxvals)
 NPY_NO_EXPORT int
 PyArray_IntpFromSequence(PyObject *seq, npy_intp *vals, int maxvals)
 {
-    return PyArray_IntpFromIndexSequence(seq, vals, (npy_intp)maxvals);
+    if (PyLong_CheckExact(seq) || !PySequence_Check(seq)) {
+        vals[0] = intp_from_scalar(seq);
+        if (PyErr_Occurred() != NULL) {
```

Comment:
```suggestion
        if (error_converting(vals[0])) {
```
Python (and we) ensure the result is -1 if an error occurred, `error_converting` uses that for the common-case.